database/agents.go

```diff
@@ -117,8 +117,14 @@ func (db *DB) InsertAgent(agt mig.Agent) (err error) {
 // UpdateAgentHeartbeat updates the heartbeat timestamp of an agent in the database
 // unless the agent has been marked as destroyed or upgraded
 func (db *DB) UpdateAgentHeartbeat(agt mig.Agent) (err error) {
-	_, err = db.c.Exec(`UPDATE agents SET status=$1, heartbeattime=$2 WHERE id=$3`,
-		mig.AgtStatusOnline, agt.HeartBeatTS, agt.ID)
+	jEnv, err := json.Marshal(agt.Env)
+	if err != nil {
+		err = fmt.Errorf("Failed to marshal agent environment: '%v'", err)
+		return
+	}
+	_, err = db.c.Exec(`UPDATE agents SET status=$1, heartbeattime=$2,
+		environment=$3, name=$4 WHERE id=$5`,
+		mig.AgtStatusOnline, agt.HeartBeatTS, jEnv, agt.Name, agt.ID)
```

Comment:
r-
This is going to overwrite existing environments. We need to keep history (useful for forensics).
It would be useful for the agent to send a flag alongside the heartbeat when its environment has changed, then the scheduler would know to store a new row instead of updating the existing one.